### PR TITLE
Adds manager.sh logs command (and other QOL improvements)

### PR DESCRIPTION
**Pull request by mycarrysun**
- Ignores downloaded steam files since we don't want to commit those
- Ignores backup files since we don't want to commit those
- Adds a `manager.sh logs` command

### CODE REVIEW

**Comment by azixus** [comment]
> * Ignores backup files since we don't want to commit those
> * Simplifies the `manager.sh` script to always use `docker compose`
>   
>   * Tested that this works by changing the .env to have the ServerName=Test Session Name and restarted the server with `docker compose down && docker compose up -d`. Output below:

I'd rather keep using the docker compose + docker trick. The main advantage is that it allows to update environment variables without having to restart the docker container. While it is indeed less "docker-y", I think the advantage is worth it.


The rest of the changes look good, just need to fix the `docker_cmd` function 👍

**Comment by mycarrysun** [comment]
> > * Ignores backup files since we don't want to commit those
> > * Simplifies the `manager.sh` script to always use `docker compose`
> >   
> >   * Tested that this works by changing the .env to have the ServerName=Test Session Name and restarted the server with `docker compose down && docker compose up -d`. Output below:
> 
> I'd rather keep using the docker compose + docker trick. The main advantage is that it allows to update environment variables without having to restart the docker container. While it is indeed less "docker-y", I think the advantage is worth it.
> 
> The rest of the changes look good, just need to fix the `docker_cmd` function 👍

If that is correct then the command that actually matters is `docker compose up -d` which is not done from within the `manager.sh` file. Since this is where the container is first started and the server is automatically started with whatever is in `.env`. Maybe that's the whole reason why in my other issue it seemed like I couldn't actually use the `manager.sh` and had to just keep doing `docker compose down && docker compose up -d`?

Btw I haven't tested the behavior you want and can do another test when I get to my other computer later, I'm thinking the `env_file` will be evaluated during `docker compose exec` and not when the container started but I may be wrong.

**Comment by azixus** [comment]
> If that is correct then the command that actually matters is `docker compose up -d` which is not done from within the `manager.sh` file. Since this is where the container is first started and the server is automatically started with whatever is in `.env`. Maybe that's the whole reason why in my other issue it seemed like I couldn't actually use the `manager.sh` and had to just keep doing `docker compose down && docker compose up -d`?

Are you referring to issue #33? I haven't been able to test it yet. Normally after the docker container is started it should work seemlessy.

> Btw I haven't tested the behavior you want and can do another test when I get to my other computer later, I'm thinking the `env_file` will be evaluated during `docker compose exec` and not when the container started but I may be wrong.

So during my testing, I discovered that `docker compose --env-file .env exec` and `docker exec --env-file .env` do not have the same behaviour. Essentially, the former does not overwrite the environment variables that were used during the creation of the container. The latter overwrites the variable, hence why I decided to use it instead. See below.
```bash
$ docker compose --env-file .env exec -it asa_server sh -c 'echo $SESSION_NAME'
My Awesome ASA Server
$ docker exec --env-file .env -it $(docker compose ps -q asa_server) sh -c 'echo $SESSION_NAME'
My Awesome ASA Server Updated
```

File: manager.sh [hunk 1]
```diff
@@ -1,14 +1,22 @@
 #!/bin/bash
-# Simple wrapper for docker commands to make life easier.
 set -euo pipefail
 
-script_dir="$(dirname "$(realpath "$0")")"
+script_dir="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
```
Comment (mycarrysun) [inline manager.sh:4]: This is a more robust/reliable way to get the script's directory, just fyi

**Comment by mycarrysun** [comment]
@azixus thanks for doing that test for me! Seems like you're right - I went back to using `docker exec` how it was before. Let me know if there's any other changes needed.

**Comment by azixus** [comment]
@mycarrysun I've just updated `manager.sh` to use the following syntax to access `sudo_cmd`: `${sudo_cmd[@]}`.
It could throw an error in some cases otherwise.

Looks good now, thanks

**Pull request merged**